cloudformation.yaml

```diff
@@ -0,0 +1,194 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: support-admin-console
+Parameters:
+  VpcId:
+    Type: String
+    Description: VpcId of your existing Virtual Private Cloud (VPC)
+    Default: vpc-e6e00183
+  Subnets:
+    Type: CommaDelimitedList
+    Description: The list of SubnetIds in your Virtual Private Cloud (VPC)
+    Default: subnet-cb91ae8d, subnet-a7b74ac2, subnet-179e8063
+  Stack:
+    Description: Applied directly as a tag
+    Type: String
+  Stage:
+    Description: Applied directly as a tag
+    Type: String
+  App:
+    Description: Applied directly as a tag
+    Type: String
+    Default: admin-console
+  AMI:
+    Description: AMI ID (may be replaced by RiffRaff for latest baked AMI)
+    Type: String
+    Default: ami-0eb88168
+  CertificateARN:
+    Description: SSL certificate ARN
+    Type: String
+
+Resources:
+  AutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      AvailabilityZones: !GetAZs
+      VPCZoneIdentifier: !Ref Subnets
+      LaunchConfigurationName: !Ref LaunchConfig
+      MinSize: 1
+      MaxSize: 2
+      TargetGroupARNs:
+      - !Ref TargetGroup
+      HealthCheckType: ELB
+      HealthCheckGracePeriod: 300
+      Tags:
+      - Key: Stack
+        Value: !Ref Stack
+        PropagateAtLaunch: true
+      - Key: App
+        Value: !Ref App
+        PropagateAtLaunch: true
+      - Key: Stage
+        Value: !Ref Stage
+        PropagateAtLaunch: true
+  LaunchConfig:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      ImageId: !Ref AMI
+      SecurityGroups:
+      - !Ref InstanceSecurityGroup
+      InstanceType: t2.micro
+      IamInstanceProfile: !Ref InstanceProfile
+      AssociatePublicIpAddress: true
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash -ev
+          aws --region ${AWS::Region} s3 cp s3://membership-dist/${Stack}/${Stage}/${App}/support-admin-console_1.0-SNAPSHOT_all.deb /tmp
+          dpkg -i /tmp/support-admin-console_1.0-SNAPSHOT_all.deb
+          /opt/cloudwatch-logs/configure-logs application ${Stack} ${Stage} ${App} /var/log/support-admin-console/application.log
+  AppRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: "/"
+      ManagedPolicyArns:
+      - !Sub arn:aws:iam::${AWS::AccountId}:policy/guardian-ec2-role-for-ssm
+      AssumeRolePolicyDocument:
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - ec2.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Policies:
+      - PolicyName: PrivateBucket
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Action: s3:GetObject
+            Resource: !Sub arn:aws:s3:::membership-private/${Stage}/support-admin-console.private.conf
```

Comment:
personally I'd much rather use secrets manager/parameter store rather than a private config file in S3 - the parameter store has worked very well for us in the Payment API, and secrets manager is even better I believe (allows for auto-rotation and such)